src/queue.hpp

```diff
@@ -821,12 +823,16 @@ struct cvk_command_batch : public cvk_command {
             m_queue->command_pool_lock();
         }
 
+        cl_int ret = cmd->build(*m_command_buffer);
+        if (ret != CL_SUCCESS) {
+            m_queue->command_pool_unlock();
```

Comment:
It looks like we could use a command pool lock holder type thing for functions like this one. Not sure how that'd have to interact with `cvk_command_buffer::begin` though.